config/default.yml

```diff
@@ -536,6 +536,12 @@ RSpec/SingleArgumentMessageChain:
   VersionChanged: '1.10'
   StyleGuide: https://www.rubydoc.info/gems/rubocop-rspec/RuboCop/Cop/RSpec/SingleArgumentMessageChain
 
+RSpec/StubbedMock:
+  Description: Checks that message expectations are not combined with message stubs.
+  Enabled: true
```

Comment:
When I wrote the first version of this cop, there were a few comments saying that it should probably be disabled by default. What do you think @pirj? How many offenses are encountered in your real world specs collection?

Reply:
I like this method more and more. Just spotted `undefined method `end_pos' for nil:NilClass` for several files there.

47162 files inspected, 7321 offenses detected. Quite a lot.
Just a couple of randomly picked ones reveal issues:

```ruby
# 24pullrequests/spec/services/popularity_scorer_spec.rb:25:42:
  let(:project) { FactoryBot.create(:project) }

    it 'returns a score of 3 if there are 5 recent issues' do
      issue = double(:issue)
      expect(project).to receive(:repo).and_return(double(updated_at: 6.months.ago))
      expect(project).to receive(:issues).and_return([issue] * 5)
      expect(popularity_scorer.score).to eq(3)
    end
```
Should be `allow`.

```ruby
    it "does not download the Cask per default" do
      allow(Cask::CaskLoader).to receive(:load).and_return(cask)
      expect(Cask::Auditor).to receive(:audit)
        .with(cask, audit_download:        false,
                    audit_appcast:         false,
                    audit_token_conflicts: false,
                    audit_new_cask:        false,
                    audit_online:          false,
                    audit_strict:          false,
                    quarantine:            true)
        .and_return(result)

      described_class.run("casktoken")
    end
```
`.and_return` is very confusing here.

Reply:
There are several offences in `rspec-expectations` itself.
Will you be happy to merge this as enabled by default if I'll manage to push through the pull request that fixes those offences?